### PR TITLE
#120 sql task skew detect

### PR DESCRIPTION
**Pull request by gyustorm**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala [hunk 1]
```diff
@@ -121,6 +124,39 @@ class SQLAppStatusListener(
...
+      import InternalAccumulator._
+      stageIds.foreach(stageId => {
+        Option(stageMetrics.get(stageId)).foreach(metrics => {
+          val taskMetrics = metrics.taskMetrics
+          if(taskMetrics == null || taskMetrics.isEmpty) {
+            return
+          }
+          val shuffleReadRecords = taskMetrics.values().asScala
+            .filter(_.names.contains(shuffleRead.RECORDS_READ))
+            .flatMap{ m =>
```
Comment (zheniantoushipashi) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala:141]: 这里要保证整个逻辑抛异常出来不能把 spark 搞挂，看要不要catch 一下

File: sql/core/src/main/java/org/apache/spark/sql/execution/ui/SQLAppStatistic.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    for (int i = 0; i < n; i++) {
+      double delta = arr[i] - miu;
+      sum += Math.pow(delta, 4);
+    }
+
+    double sigma = standardDeviation(arr, n);
+    return (sum / (n * Math.pow(sigma, 4))) - 3;
+  }
+
+  public static void main(String[] args) {
```
Comment (jiaoew1991) [inline sql/core/src/main/java/org/apache/spark/sql/execution/ui/SQLAppStatistic.java:81]: 不要用main测试，加到ut里

File: sql/core/src/main/java/org/apache/spark/sql/execution/ui/SQLAppStatistic.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+/**
+ * https://www.itl.nist.gov/div898/handbook/eda/section3/eda35b.htm
+ */
+public class SQLAppStatistic {
+
+  static final double STANDARD_SKEWNESS = 1.0d;
+  static final double STANDARD_KURTOSIS = 1.0d;
+
+  // mean of data.
+  static double mean(double arr[], int n) {
```
Comment (jiaoew1991) [inline sql/core/src/main/java/org/apache/spark/sql/execution/ui/SQLAppStatistic.java:29]: double[] arr not double arr[]

**Comment by gyustorm** [comment]
retest this please

**Comment by gyustorm** [comment]
retest this please

**Pull request merged**